tests/testsuite/build.rs

```diff
@@ -4724,3 +4724,51 @@ d
         )
         .run();
 }
+
+#[cargo_test]
+fn simple_terminal_width() {
+    if !is_nightly() {
+        // --terminal-width is unstable
+        return;
+    }
+    let p = project()
+        .file(
+            "src/lib.rs",
+            "
+                fn a() {
+/**/                    /**/ \
+/**/                    /**/ \
+/**/                    /**/ \
+/**/                    /**/ \
+/**/                    /**/ \
+/**/                    /**/ \
+/**/                    /**/ \
+                    let _: () = 42;
+                }
+            ",
+        )
+        .build();
+    let expected = "   Compiling foo v0.0.1 (/Users/ekuber/workspace/cargo/target/cit/t0/foo)
```

Comment:
We have some macros to make matching more reliable.  For example, the first line could be 
```
[COMPILING] foo v0.0.1 ([..])
```

Also, I'm concerned about comparing the entire output here.  Since it changes fairly regularly, it might be good to just use `with_stderr_contains` instead, and just checking for the important part (like the ellipses line).